src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeIndex_Persistence.cs

```diff
@@ -19,21 +20,25 @@ internal sealed partial class SyntaxTreeIndex : IObjectWritable
 
         public readonly Checksum Checksum;
 
-        private static async Task<SyntaxTreeIndex?> LoadAsync(
-            Document document, Checksum checksum, CancellationToken cancellationToken)
-        {
-            var solution = document.Project.Solution;
-            var persistentStorageService = (IChecksummedPersistentStorageService)solution.Workspace.Services.GetRequiredService<IPersistentStorageService>();
+        private static Task<SyntaxTreeIndex?> LoadAsync(Document document, Checksum checksum, CancellationToken cancellationToken)
+            => LoadAsync(document.Project.Solution.Workspace, DocumentKey.ToDocumentKey(document), checksum, GetStringTable(document.Project), cancellationToken);
 
+        public static async Task<SyntaxTreeIndex?> LoadAsync(
```

Comment:
should this take in the storage service instead of the workspace?  iirc Tomas was wanting to remove workspace references from the OOP, so moving the workspace references up to the top might make it easier to remove later on down the line.

Reply:
i think that will take a lot of work no matter what.  so i'd be ok just waiting until that stage.